### PR TITLE
Improve the `wasmtime` crate's README

### LINKED ISSUE #3704: update JIT/AOT mentions in the repo
Now that wasmtime supports AOT I suggest you update the "about" section of this project as well as the tags.

For context, I didn't realize that wasmtime was supporting AOT until I went to the Lucet repo where they mentioned AOT support in wasmtime! It might be nice to mention that on the website and README, since the only mention I found was buried there: https://docs.wasmtime.dev/cli-options.html#compile

### LINKED ISSUE #3707: Documentation Improvement: wasmtime crate readme page
The [`wasmtime` crate](https://crates.io/crates/wasmtime)'s readme is a bit sparse and isn't as thorough as the [Rust - Wasmtime](https://docs.wasmtime.dev/lang-rust.html) documentation page. Although it doesn't need to be (since they serve different purposes), it can and probably should be enhanced. By improving this page `wasmtime` can make itself more appealing and accessible to users looking at WebAssembly projects on crates.io.

One additional point of comparison is to the [`wasmer` crate readme](https://crates.io/crates/wasmer), which includes some features that can be borrowed.
1. Using links for anything that has an associated page (e.g. on "`wasmtime` Webassembly runtime")
2. Offering a "usage" section with an example and links to where you can find more.
3. Offering a "features" section that describes any high-level features of the crate.
4. The inclusion of CI, crates.io version, and license badges.

While a bit more nebulous, the `wasmer` crate also has a much warmer feeling to it and makes a lot of appeals to users like
* mentioning that it is "the most popular WebAssembly runtime",
* emphasizing that "It's designed to be safe and secure, and runnable in any kind of environment",
* saying that wasmer is "Made with ❤️ by the Wasmer team, for the community".

While it doesn't make sense for wasmtime to try to match these various appeals 1-to-1, it might make sense to think about what messages about wasmtime the crates page should try to convey to users about the tool, it's properties, and it's creators. One appeal that might make sense is to refer to wasmtime's status as part of the Bytecode Alliance.

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit is me finally getting back to #2688 and improving the README
of the `wasmtime` crate. Currently we have a [pretty drab README][drab]
that doesn't really convey what we want about Wasmtime.

While I was doing this I opted to update the feature list of Wasmtime as
well in the main README (which is mirrored into the crate readme),
namely adding a bullet point for "secure" which I felt was missing
relative to how we think about Wasmtime.

Naturally there's a lot of ways to paint this shed, so feedback is of
course welcome on this! (I'm not the best writer myself)

Closes #3704
Closes #3707

[drab]: https://crates.io/crates/wasmtime/0.37.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,16 +62,21 @@ Hello, world!
...
-  wasm-executing needs.
+* **Secure**. Wasmtime's development is strongly focused on the correctness of
+  its implementation with 24/7 fuzzing donated by [Google's OSS Fuzz],
+  leveraging Rust's API and runtime safety guarantees, careful design of
+  features and APIs through an [RFC process], a [security policy] in place
+  for when things go wrong, and a [release policy] for patching older versions
+  as well.
+
+* **Configurable**. Wastime supports a rich set of APIs and build time
+  configuration to provide many option such as further means of restricting
```
Comment (abrown) [inline README.md:76]:
```suggestion
  configuration to provide many options such as further means of restricting
```

**Comment by cfallin** [comment]
Re: security -- if we can point to current efforts in addition to things-that-fully-exist-today, we could mention "We're also working with academic collaborators to fully formally verify our compilation pipeline", maybe?

It might also be worth mentioning that we follow best practices for security wrt guest code sandboxing, e.g. Spectre mitigations, and that we try to adopt defense-in-depth strategies where we can (extra guard regions, zeroing memory immediately after termination, using hardware CFI features to guard control flow, ...).

File: README.md [hunk 2]
```diff
@@ -62,16 +62,21 @@ Hello, world!
...
 
-* **Configurable**. Whether you need to precompile your wasm ahead of time,
-  or interpret it at runtime, Wasmtime has you covered for all your
-  wasm-executing needs.
+* **Secure**. Wasmtime's development is strongly focused on the correctness of
+  its implementation with 24/7 fuzzing donated by [Google's OSS Fuzz],
+  leveraging Rust's API and runtime safety guarantees, careful design of
+  features and APIs through an [RFC process], a [security policy] in place
+  for when things go wrong, and a [release policy] for patching older versions
+  as well.
```
Comment (cfallin) [inline README.md:73]:
```suggestion
as well. We follow best practices for defense-in-depth and known
protections and mitigations for issues like Spectre. Finally, we're working
to push the state-of-the-art: we are collaborating with academic
researchers to formally verify critical parts of Wasmtime and Cranelift.
```

**Review by cfallin — APPROVED** [review]
Thanks for improving this README situation!

Suggestion included below for the thought above but feel free to edit or trim as needed...

**Comment by alexcrichton** [comment]
I opted to also go ahead and merge `security-sandboxing.md` with `security.md` since `security.md` was otherwise empty. I additionall expanded the documentation with items you mentioned @cfallin. Finally I also tried to update some various documentation for #3704 to make sure relevant locations mention ahead-of-time where appropriate.

**Review by cfallin — APPROVED** [review]
New `security.md` content looks great, thanks a ton for writing that!

**Pull request merged**